lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java

```diff
@@ -510,14 +510,50 @@ private void removeSingleTaskForCallbackStateByExternalId(BulkOperationResults<S
                 bulkLog.addError(currentExternalId, new TaskNotFoundException(currentExternalId,
                     TASK_WITH_ID + currentExternalId + WAS_NOT_FOUND2));
                 externalIdIterator.remove();
-            } else if (!TaskState.COMPLETED.equals(foundSummary.getTaskState())) {
+            } else if (!desiredCallbackStateCanBeSetForFoundSummary(foundSummary, desiredCallbackState)) {
                 bulkLog.addError(currentExternalId, new InvalidStateException(currentExternalId));
                 externalIdIterator.remove();
             }
+
         }
         LOGGER.debug("exit from removeSingleTask()");
     }
 
+    private boolean desiredCallbackStateCanBeSetForFoundSummary(MinimalTaskSummary foundSummary, CallbackState desiredCallbackState) {
```

Comment:
this is such a good case of unit testing domain logic. please consider making this package-private and adding a few tests to verify this logic.

Reply:
The acceptance test includes full code coverage of this logic. Since there will be a refactoring of the TaskServiceImpl class very soon, any unit tests would have to be adjusted again, while the acceptance tests stays as it is. For these reasons I won't add extra unit tests for now.